components/payments/internal/connectors/engine/engine.go

```diff
@@ -153,7 +158,51 @@ func (e *engine) UninstallConnector(ctx context.Context, connectorID models.Conn
 	return nil
 }
 
-func (e *engine) CreateBankAccount(ctx context.Context, bankAccountID uuid.UUID, connectorID models.ConnectorID) (*models.BankAccount, error) {
+func (e *engine) ResetConnector(ctx context.Context, connectorID models.ConnectorID) error {
+	connector, err := e.storage.ConnectorsGet(ctx, connectorID)
+	if err != nil {
+		return err
+	}
+
+	// Detached the context to avoid being in a weird state if request is
+	// cancelled in the middle of the operation.
+	ctx, _ = contextutil.Detached(ctx)
```

Comment:
If the work is detached from the parent context we'll also need to create an internal waitgroup that can advertise to main that engine events such as this one are finished so that it can execute a graceful shutdown (similar to how http.Server.Shutdown() is implemented).

(I suggest doing this is a separate ticket though)

Reply:
Yes you're right, let me add a ticket for this one !